proc_scripts/align_slice.py

```diff
@@ -28,27 +28,32 @@ def align_and_slice(s, dimension='t2', convwidth=500, threshold=0.05, fl=None):
     if fl is not None:
         fl.pop_marker()
     return s[dimension:slices]
-def correlation_align(s,avg,convwidth=0,dimension='t2',fl=None):
-    assert not s.get_ft_prop(dimension), 'I want time-domain data'
-    assert not avg.get_ft_prop(dimension), 'I want time-domain data'
+def correlation_align(s,avg,convwidth=0,axis='t2',fl=None):
+    assert not s.get_ft_prop(axis), 'I want time-domain data'
+    assert not avg.get_ft_prop(axis), 'I want time-domain data'
     avg.run(conj)
     R = pi*convwidth # Lorentzian FWHM formula -- R = 1/T2, FWHM = R/pi
     # ALSO convolve both
-    check = avg * s * exp(-2*R*abs(s.fromaxis(dimension)))
-    assert s.get_ft_prop(dimension,['start','freq']) is not None, "should be FT'd first, so the startpoint is set"
-    check.ft('t2', pad=2**14)
-    check.reorder('t2',first=False)
-    forplot = abs(check)['t2':(-50,50)]
-    myline = forplot.C.argmax('t2')
-    forplot.smoosh(['ph1','ph2','power'],'indirect'
+    check = avg * s * exp(-2*R*abs(s.fromaxis(axis)))
+    assert s.get_ft_prop(axis,['start','freq']) is not None, "should be FT'd first, so the startpoint is set"
+    check.ft(axis, pad=2**14)
+    check.reorder(axis,first=False)
+    forplot = abs(check)[axis:(-50,50)]
+    myline = forplot.C.argmax(axis)
+    indirect_dims = set(s.dimlabels) - set([axis])
+    phcyc_dims = [j for j in indirect_dims if j.startswith('ph')]
+    phcyc_dims.sort()
+    indirect_dims = list(set(indirect_dims) - set(phcyc_dims))
+    forplot.reorder(phcyc_dims+indirect_dims)
+    forplot.smoosh(phcyc_dims+indirect_dims,'indirect'
             ).setaxis('indirect','#'
                     ).reorder('indirect',first=False)
     if fl is not None:
         fl.push_marker()
         fl.next('cross-correlation')
-        fl.image(forplot,human_units=True)
-        fl.plot(myline.C.smoosh(['ph1','ph2','power'],'indirect'
+        fl.image(forplot)
+        fl.plot(myline.smoosh(phcyc_dims+indirect_dims,'indirect'
             ).setaxis('indirect','#'
                     ).reorder('indirect',first=False)
                 , 'w', linewidth=3, alpha=0.5)
-    return s*exp(-1j*2*pi*myline*s.fromaxis('t2'))
+    return s*exp(-1j*2*pi*myline*s.fromaxis(axis))
```

Comment:
these edits are in response to [this comment](https://github.com/jmfrancklab/proc_scripts/pull/13#discussion_r434751630)